tests/test_utils.py

```diff
@@ -66,15 +66,15 @@ def test_run_cmd_captures_stderr(self):
             'print("Hello", file=sys.stderr)',
         ]
         self.assertEqual(utils.run_cmd(cmd), '')
-        with self.assertRaisesRegexp(SubcommandException, r'STDERR[\s-]+Hello'):
+        with self.assertRaisesRegex(SubcommandException, r'STDERR[\s-]+Hello'):
             cmd[-1] += '; sys.exit(99)'
             utils.run_cmd(cmd)
 
     def test_pwd_git_version_without_git(self):
         # Change into test dir to make sure we're within a repo
         os.chdir(os.path.dirname(__file__))
         self.assertIsNotNone(utils.pwd_git_version())
-        with patch('shub.utils.find_executable', return_value=None):
+        with patch('shub.utils.which', return_value=None):
```

Comment:
Should this be `shub.utils.which` or `shitil.which`?

Reply:
This confused me at first too, but because `shub.utils` is already imported before applying the `patch` we have to use the [same object name](https://docs.python.org/3/library/unittest.mock.html#where-to-patch
) that `pwd_git_version` refers to (other tests also do this, see `@patch('shub.utils.time.time')` right below)